src/test/java/com/teragrep/pth10/SyslogStreamTest.java

```diff
@@ -95,74 +115,108 @@ void setEnv() {
-            arrayOfSyslogs.set(numberOfSyslogMessagesSent.getAndIncrement(), new String(message));
-        };
-
-        final int port = 9999;
-        final Server server = new Server(port, new SyslogFrameProcessor(cbFunction));
-        Assertions.assertDoesNotThrow(server::start);
-
         streamingTestUtil
                 .performDPLTest(
-                        "index=index_A | teragrep exec syslog stream host 127.0.0.1 port " + port, testFile, ds -> {
-                            LOGGER.debug("Syslog msgs = <{}>", numberOfSyslogMessagesSent.get());
-                            Assertions.assertEquals(expectedSyslogs, numberOfSyslogMessagesSent.get());
-
-                            for (int i = 0; i < expectedSyslogs; i++) {
-                                String s = arrayOfSyslogs.get(i);
-                                for (int j = 0; j < expectedSyslogs; j++) {
-                                    if (i == j)
-                                        continue;
-                                    Assertions.assertFalse(arrayOfSyslogs.compareAndSet(j, s, s));
-                                }
-
-                            }
-                            Assertions.assertAll("stop server", server::stop);
+                        "index=index_A | teragrep exec syslog stream host 127.0.0.1 port " + listenPort, testFile,
+                        ds -> {
+                            LOGGER.debug("Syslog msgs = <{}>", messages.size());
+                            Assertions.assertEquals(10, messages.size());
+                            Assertions.assertEquals(10, new HashSet<>(messages).size());
                         }
                 );
     }
 
-    @Disabled(value = "RLP-03 has to be updated") // FIXME: update rlp_03
     @Test
     @DisabledIfSystemProperty(
             named = "skipSparkTest",
             matches = "true"
     ) // teragrep exec syslog stream, with preceding aggregation command
     public void syslogStreamSendingFailureTest() {
-        Assertions
-                .assertThrows(
-                        StreamingQueryException.class,
-                        () -> streamingTestUtil
-                                .performDPLTest(
-                                        "index=index_A | stats count(_raw) as craw | teragrep exec syslog stream host 127.0.0.1 port 9998",
-                                        testFile, ds -> {
-                                        }
-                                )
+        RuntimeException rte = streamingTestUtil
+                .performThrowingDPLTest(
+                        RuntimeException.class,
+                        "index=index_A | stats count(_raw) as craw | teragrep exec syslog stream host 127.1.0.1 port 9998",
+                        testFile, ds -> {
+                        }
                 );
+
+        Assertions.assertNotNull(rte);
+        Assertions.assertTrue(rte.getMessage().contains("cannot be used after aggregations!"));
```

Comment:
assertEquals for the whole Exception message would be better. I can't see from this if the message is readable in the `"'Step ' + step + ' cannot be used...'"` format.

Reply:
fixed